### PR TITLE
Make the Upstart script consistent with the systemd script

### PR DESCRIPTION
**Pull request by bobbrez**

### CODE REVIEW

**Comment by bobbrez** [comment]
I signed the CLA, this should be ready to go.

File: templates/init/debian/elasticsearch.j2 [hunk 1]
```diff
@@ -103,7 +103,7 @@ fi
 
 # Define other required variables
 PID_FILE="$PID_DIR/$NAME.pid"
-DAEMON=$ES_HOME/bin/elasticsearch
+DAEMON={{es_home}}/bin/elasticsearch
```
Comment (strootman) [inline templates/init/debian/elasticsearch.j2]: I would have left this part be. The templating at line 52 takes care of this.

Comment (bobbrez) [inline templates/init/debian/elasticsearch.j2]: With `{{es_home}}` it makes it consistent with how the [existing systemd](https://github.com/elastic/ansible-elasticsearch/blob/master/templates/systemd/elasticsearch.j2#L18) is handling it. I'm fine either way.

**Comment by gingerwizard** [comment]
@bobbrez As far as i can tell none of this is needed as the script reads the default file for its environment settings.  Happy to merge for consistency however althoigh functionally im not sure it brings anything

**Pull request merged**